src/Factory.php

```diff
@@ -87,11 +88,23 @@ public function __call(string $name, array $arguments): array
     }
 
     /**
-     * @return Proxy<TObject>&TObject
-     * @phpstan-return Proxy<TObject>
+     * @return (Proxy<TObject>&TObject)|TObject
+     * @phpstan-return ($noProxy is true ? TObject: Proxy<TObject>)
+     * @psalm-suppress ImplementedReturnTypeMismatch
      */
-    final public function create(array|callable $attributes = []): Proxy
+    final public function create(
+        array|callable $attributes = [],
+        /**
+         * @deprecated
+         * @internal
+         */
+        bool $noProxy = false
+    ): object
     {
+        if (\count(func_get_args()) === 2 && !str_starts_with(debug_backtrace(options: \DEBUG_BACKTRACE_IGNORE_ARGS, limit: 2)[1]['class'] ?? '', 'Zenstruck\Foundry')) {
```

Comment:
Cool solution!

Reply:
I'm wondering if I won't remove the other `calledInternally` parameter in favor of this